File: src/Entity/Metier.php
<?php

namespace App\Entity;

use App\Repository\MetierRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: MetierRepository::class)]
class Metier
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: 'Entrer nom metier')]
    #[Assert\Length(
        min: 5,
        max: 255,
        minMessage: 'The name must be at least {{ limit }} characters',
        maxMessage: 'The name cannot exceed {{ limit }} characters'
    )]
    #[Assert\Regex("/^[a-zA-Z]/")]
    private ?string $nom = null;

    #[ORM\Column(length: 255)]
    #[Assert\NotBlank(message: 'Entrer type metier')]
    #[Assert\Length(
        min: 5,
        max: 255,
        minMessage: 'The name must be at least {{ limit }} characters',
        maxMessage: 'The name cannot exceed {{ limit }} characters'
    )]
    #[Assert\Regex("/^[a-zA-Z]/")]
    private ?string $type = null;

    #[ORM\Column(type: Types::TEXT,unique:true)]
    #[Assert\NotBlank(message: 'Entrer description metier')]
    #[Assert\Length(
        min: 5,
        max: 255,
        minMessage: 'The name must be at least {{ limit }} characters',
        maxMessage: 'The name cannot exceed {{ limit }} characters'
    )]
    #[Assert\Regex("/^[a-zA-Z]/")]

    private ?string $description = null;

    #[ORM\Column(length: 255,nullable: true)]
    private ?string $image = null;

    #[ORM\Column(nullable: true)]
    private ?bool $archive = null;

    #[ORM\OneToMany(mappedBy: 'Metier', targetEntity: SousMetier::class)]
    private Collection $sousMetiers;

   

    public function __construct()
    {
        $this->sousMetiers = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): self
    {
        $this->type = $type;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): self
    {
        $this->image = $image;

        return $this;
    }

    public function isArchive(): ?bool
    {
        return $this->archive;
    }

    public function setArchive(?bool $archive): self
    {
        $this->archive = $archive;

        return $this;
    }

    /**
     * @return Collection<int, SousMetier>
     */
    public function getSousMetiers(): Collection
    {
        return $this->sousMetiers;
    }

    public function addSousMetier(SousMetier $sousMetier): self
    {
        if (!$this->sousMetiers->contains($sousMetier)) {
            $this->sousMetiers->add($sousMetier);
            $sousMetier->setMetier($this);
        }

        return $this;
    }

    public function removeSousMetier(SousMetier $sousMetier): self
    {
        if ($this->sousMetiers->removeElement($sousMetier)) {
            // set the owning side to null (unless already changed)
            if ($sousMetier->getMetier() === $this) {
                $sousMetier->setMetier(null);
            }
        }

        return $this;
    }
    public function __toString()
    {
        return $this->nom;
    }

  
}

File: src/com/mycomany/entities/Metier.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycomany.entities;



/**
 *
 * @author Safe
 */
public class Metier {
    private String nom,type,description,image;
    private int id;
    
    public Metier(){};
    public Metier(int id){
    this.id = id;
    }

    public Metier(String nom, String type, String description, String image, int id) {
        this.nom = nom;
        this.type = type;
        this.description = description;
        this.image = image;
        this.id = id;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }
    
    public Metier(String nom, String type, String description) {
        this.nom = nom;
        this.type = type;
        this.description = description;
    }
     public Metier(String nom) {
        this.nom = nom;
       
    }
     public Metier(int id,String nom, String type, String description) {
        this.id = id ;
        this.nom = nom;
        this.type = type;
        this.description = description;
    }
    public void setId(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }
   

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setType(String type) {
        this.type = type;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getNom() {
        return nom;
    }

    public String getType() {
        return type;
    }

    public String getDescription() {
        return description;
    }


    
    @Override
    public String toString() {
        return "Metier{"+"id=" + id  + ", nom=" + nom + ", type=" + type + ", description=" + description + '}';
    }
    
}
